nbconvert/preprocessors/regexremove.py

```diff
@@ -39,19 +40,36 @@ class RegexRemovePreprocessor(Preprocessor):
 
     patterns = List(Unicode, default_value=[r'\Z']).tag(config=True)
 
-    def preprocess(self, nb, resources):
+    def check_conditions(self, cell):
         """
-        Preprocessing to apply to each notebook. See base.py for details.
+        Checks that a cell matches the pattern and that (if a code cell)
```

Comment:
The comment here doesn't quite match the code below: The function checks that the cell does _not_ match the pattern _or_ is a code cell with output.

As an aside, are there any other cell types that can have output? We might just be able to check `not pattern.match(cell.source) or cell.get('outputs')`. Then `cell.get('outputs')` will evaluate to false if the key doesn't exist or the list is empty.